File: clients/src/main/java/org/yigitcanyontem/clients/content/dto/TopicDto.java
package org.yigitcanyontem.clients.content.dto;

import lombok.*;

import java.util.Date;
import java.util.Set;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class TopicDto {
    private Long id;

    private Long parentTopicId;

    private String name;

    private String description;

    private Integer createdByUserId;

    private String createdByUsername;

    private Integer updatedByUserId;

    private String updatedByUsername;

    private Date createdAt;

    private Long viewCountTotal;

    private Long viewCountLastWeek;

    private String slug;

    private Set<TagDto> tags;

    private Long replyCount;

    private Long childTopicCount;
}

File: frontend/codeConnect/src/models/content/TopicDto.ts
import {TagDto} from "@/models/content/TagDto.ts";

export interface TopicDto {
    id: number;
    parentTopicId: number;
    name: string;
    description: string;
    createdByUserId: number;
    createdByUsername: string;
    updatedByUserId: number;
    updatedAt: Date;
    updatedByUsername: string;
    createdAt: Date;
    viewCountTotal: number;
    viewCountLastWeek: number;
    slug: string;
    tags: Set<TagDto>;
    replyCount: number;
}
